aggregate_logging_sizing.adoc

```diff
@@ -0,0 +1,283 @@
+
+
+== Storage Considerations 
+An Elasticsearch (ES) index is a collection of shards and it's correspondent replica shards. This is how ES implements high availability internally, therefore there is little need of using hardware based mirroring RAID variants. RAID 0 can still be used to increase overall disk performance. +
+Every search request needs to hit a copy of every shard in the index, which is fine if every shard is sitting on a different node, but not if many shards have to compete for the same resources.
+Each ES instance requires it's own individual storage, but an OpenShift deployment can only provide volumes shared by all of its pods, which again means that Elastic shouldn't be implemented with a single master. +
+
+A persistent storage volume should be added to each Elasticsearch deployment configuration so that we have one volume per https://www.elastic.co/guide/en/elasticsearch/guide/current/replica-shards.html[replica shard]. On Openshift this is often achieved through https://docs.openshift.com/enterprise/3.2/architecture/additional_concepts/storage.html#persistent-volume-claims[Persistent Volume Claims] +
+
+* 1 volume per shard
+
+* 1 volume per replica shard
+  
+The PVCs must be named based on the `es-pvc-prefix` setting. Refer to https://github.com/openshift/openshift-docs/blob/master/install_config/aggregate_logging.adoc#overview[Persistent Elasticsearch Storage] for more details. Below are some guidelines to help dimensioning your EFK deployment.
+
+*Example scenario*
+
+Assumptions: +
+1) which app: Apache +
+2) bytes per line: 256 +
+3) lines per second load on app: 1 +
+4) raw text data -> json +
+
+
+Logging project:
+
+Baseline (256 characters per minute -> 15KB/min)
+
+----
+
+3 es 
+1 kibana
+1 curator
+1 fluentd
+6 pods: 90000 x 86400 = 7,7 GB/day
+
+
+3 es
+1 kibana
+1 curator
+11 fluentd 
+16 pods: 225000 x 86400 = 24,0 GB/day
+
+3 es
+1 kibana
+1 curator
+20 fluentd 
+25 pods: 225000 x 86400 = 32,4 GB/day
+
+----
+
+A general logging load can be obtained with some simple formulas:
+
+----
+ (bytes-per-line * (lines-per-second) = 2560 bytes per app per second
+ (2560) * (number-of-pods-per-node,100) = 256,000 bytes per second per node
+ 256k * (number-of-nodes) = total logging throughput per cluster
+----
+
+Fluentd gets any logs from `/var/log/messages` and `/var/lib/docker/containers/*`
+
+Local SSD drives are recommended in order to achieve the best performance. Also, if you decide for SSD it might be convenient to change the default I/O scheduler (CFQ) since it is mostly optimized for regular spinning media. +
+Since sizing storage for ES is in great part connected to how you optimize your indexes you should think about how much data you will need in advance, keeping in mind we are aggregating application log data. 
```

Comment:
RHEL's default is deadline.  And it's important to note that in public cloud, the elevator inside the VM is largely irrelevant because the hypervisor is going to do it's own thing with those I/O's anyway.